docs/tutorial.md

```diff
@@ -0,0 +1,507 @@
+
+Next, add the `circuit-foundation` dependency. This includes all the core Circuit artifacts.
+
+=== "build.gradle.kts"
+    ```kotlin
+    dependencies {
+      implementation("com.slack.circuit:circuit-foundation:<version>")
+    }
+    ```
+
+TODO for more details see setup
+
+## Create a `Screen`
+
+The primary entry points in Circuit are `Screen`s. These are the navigational building blocks of your app. A `Screen` is a simple data class or data object that represents a unique location in your app. For example, a `Screen` could represent an inbox list, an email detail, or a settings screen.
+
+Let's start with a simple `Screen` that represents an inbox list:
+
+=== "Android"
+    ```kotlin
+    @Parcelize
+    data object InboxScreen : Screen
+    ```
+
+=== "Multiplatform"
+    ```kotlin
+    data object InboxScreen : Screen
+    ```
+
+!!! note
+    `Screen` is `Parcelable` on Android. You should use the Parcelize plugin to annotate your screens with `@Parcelize`.
+
+## Design your state
+
+Next, let's define some state for our `InboxScreen`. Circuit uses unidirectional data flow (UDF) to ensure strong separation between presentation logic and UI. States should _stable_ or _immutable_ (TODO link), and directly renderable by your UIs. As such, you should design them to be as simple as possible.
+
+Conventionally, this is written as a nested `State` class inside your `Screen` and _must_ extend `CircuitUiState`.
+
+=== "InboxScreen.kt"
+    ```kotlin
+    data object InboxScreen : Screen {
+      data class State(
+        val emails: List<Email>
+      ) : CircuitUiState
+    }
+    ```
+
+=== "Email.kt"
+    ```kotlin
+    @Immutable
+    data class Email(
+      val id: String,
+      val subject: String,
+      val body: String,
+      val sender: String,
+      val timestamp: String,
+      val recipients: List<String>,
+    )
+    ```
+
+See the [states and events](states-and-events.md) guide for more information.
+
+## Create your UI
+
+Next, let's define a `Ui` for our `InboxScreen`. A `Ui` is a simple composable function that takes `State` and `Modifier` parameters. It's responsible for rendering the state. You should write this like a standard composable. In this case, we'll use a `LazyColumn` to render a list of emails.
+
+// TODO side by side screenshot?
+
+=== "Inbox"
+    ```kotlin
+    @Composable
+    fun Inbox(state: InboxScreen.State, modifier: Modifier = Modifier) {
+      LazyColumn(modifier = modifier) {
+        items(state.emails.size) { index ->
+          EmailItem(state.emails[index])
+        }
+      }
+    }
+    
+    // Write one or use EmailItem from ui.kt
+    @Composable
+    private fun EmailItem(email: Email, modifier: Modifier = Modifier) {
+      // ...
+    }
+    ```
+
+## Implement your presenter
+
+Next, let's define a `Presenter` for our `InboxScreen`. Circuit presenters are responsible for computing and emitting state.
```

Comment:
Here (or perhaps the note below) is a good place to explain how using the Compose engine allows updates to the state to re-flow to the UI. I think that's an easy detail to miss here - especially with the source being static in the tutorial.

Reply:
[ce77205](https://github.com/slackhq/circuit/pull/1158/commits/ce77205357576e535cfc21589b9c7f5ca7a4d775)